supervision/dataset/formats/coco.py

```diff
@@ -124,7 +124,7 @@ def detections_to_coco_annotations(
             "category_id": int(class_id),
             "bbox": [xyxy[0], xyxy[1], box_width, box_height],
             "area": box_width * box_height,
-            "segmentation": polygon,
+            "segmentation": [polygon],
```

Comment:
if there is no polygon,segmentation will be `[[]]`. 

`[[]]` in python if condition will be `True`.

`[]` in python if condition will be `False`.


segmentation in coco dataset is `[]`  when there is no segmentatio which means `False` in python.Although it is a small change, we should be as cautious as possible considering that someone may make a Boolean judgment on it.

change `[polygon]` to `[polygon] if polygon else []` may avoid this problem.

Reply:
Makes a lot of sense. Apart from truth value, it could also break anything using the size of the array.  Have added to the fix